src/main/java/io/aiven/kafka/connect/opensearch/OpensearchSinkTask.java

```diff
@@ -151,15 +151,25 @@ protected String convertTopicToIndexName(final String topic) {
             indexName = indexName.replace(".", "dot");
             LOGGER.warn("Topic {} name is . or .. . The final index name is {}", topic, indexName);
         }
-        return indexName;
+        final var finalIndexName = indexName;
+        return config.dataStreamEnabled()
```

Comment:
I still don't understand why do you need to check `config.dataStreamEnabled()` on every record :(